pkg/infrastructure/aws/clusterapi/ignition.go

```diff
@@ -0,0 +1,75 @@
+package clusterapi
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/sirupsen/logrus"
+	capa "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	awstypes "github.com/openshift/installer/pkg/types/aws"
+	"github.com/openshift/installer/pkg/types/dns"
+)
+
+func editIgnition(ctx context.Context, in clusterapi.IgnitionInput) ([]byte, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*2)
```

Comment:
Here you're overriding the context you received as parameter and then you're passing it through to `clusterapi.EditIgnition` at the end of this function. Let's either: 1) not set a timeout; or 2) use separate timed contexts. For now we know that `clusterapi.EditIgnition` is ignoring the context parameter, but that could change and this detail would be missed.